gammapy/spectrum/crab.py

```diff
@@ -15,138 +15,89 @@
+        return flux / energy ** 2
 
-    See the ``gammapy.spectrum.crab`` module docstring for a description
-    of the available reference spectra.
 
-    Parameters
-    ----------
-    energy : array_like
-        Energy (TeV)
-    reference : {{'hegra', 'hess_pl', 'hess_ecpl', 'meyer'}}
-        Published Crab reference spectrum
-
-    Returns
-    -------
-    flux : array
-        Differential flux (cm^-2 s^-1 TeV^-1) at ``energy``
-    """
-    if reference == 'hegra':
-        f = hegra['diff_flux']
-        g = hegra['index']
-        return f * energy ** (-g)
-    elif reference == 'hess_pl':
-        f = hess_pl['diff_flux']
-        g = hess_pl['index']
-        return f * energy ** (-g)
-    elif reference == 'hess_ecpl':
-        f = hess_ecpl['diff_flux']
-        g = hess_ecpl['index']
-        e_c = hess_ecpl['cutoff']
-        return f * energy ** (-g) * np.exp(-energy / e_c)
-    elif reference == 'meyer':
-        # Meyer et al., 2010arXiv1008.4524M, Appendix D
-        p = np.array([-0.00449161, 0, 0.0473174,
-                      - 0.179475, -0.53616, -10.2708])
-        log_energy = np.log10(np.asarray(energy))
-        log_flux = np.poly1d(p)(log_energy)
-        flux = 10 ** log_flux
-        return Unit('erg').to('TeV') * flux / energy ** 2
-    else:
-        raise ValueError('Unknown reference: {0}'.format(reference))
-
-
-def crab_integral_flux(energy_min=1, energy_max=1e4, reference=CRAB_DEFAULT_REFERENCE):
-    """Integral Crab flux.
-
-    See the ``gammapy.spectrum.crab`` module docstring for a description
-    of the available reference spectra.
-
-    Parameters
-    ----------
-    energy_min, energy_max : array_like
-        Energy band (TeV)
-    reference : {{'hegra', 'hess_pl', 'hess_ecpl', 'meyer'}}
-        Published Crab reference spectrum
-
-    Returns
-    -------
-    flux : array
-        Integral flux (cm^-2 s^-1) in energy band ``energy_min`` to ``energy_max``
+class CrabSpectrum(object):
     """
-    from scipy.integrate import quad
-    # TODO: How does one usually handle 0-dim and 1-dim
-    # arrays at the same time?
-    energy_min, energy_max = np.asarray(energy_min, dtype=float), np.asarray(energy_max, dtype=float)
-    npoints = energy_min.size
-    energy_min, energy_max = energy_min.reshape(npoints), energy_max.reshape(npoints)
-    I, I_err = np.empty_like(energy_min), np.empty_like(energy_max)
-    for ii in range(npoints):
-        I[ii], I_err[ii] = quad(crab_flux, energy_min[ii], energy_max[ii],
-                                (reference), epsabs=1e-20)
-    return I
-
+    Crab spectral model.
 
-def crab_spectral_index(energy=1, reference=CRAB_DEFAULT_REFERENCE):
-    """Spectral index (positive number) at a given energy.
+    The following references are available:
 
-    See the ``gammapy.spectrum.crab`` module docstring for a description
-    of the available reference spectra.
+        * 'meyer', 2010arXiv1008.4524M
```

Comment:
No need to indent lists, putting them at the same level as the surrounding text gives nice HTML output.